### ISSUE TITLE
mouseHoverEvent method of ButtonItem class not work

### ISSUE THREAD

**Issue by hatawong** [pyqtgraph/pyqtgraph#2123]
### `def mouseHoverEvent(self, ev) ` of ButtonItem class not work
right method should be  `def hoverEvent(self, ev) `

**Comment by j9ac9k** [comment]
Hi @hatawong can you provide a code example showing it not working?

**Comment by hatawong** [comment]
```
import pyqtgraph as pg
from pyqtgraph import icons


class PlotItem(pg.PlotItem):

    def __init__(self, **kargs):
        super().__init__(**kargs)
        self.autoBtn.setParentItem(None)
        self.autoBtn = ButtonItem(imageFile=icons.getGraphPixmap('auto'), width=14, parentItem=self)
        self.autoBtn.mode = 'show'
        self.autoBtn.clicked.connect(self.autoBtnClicked)


class ButtonItem(pg.ButtonItem):

    def __init__(self, **kargs):
        super().__init__(**kargs)

    def mouseHoverEvent(self, ev):  # original implements - not work
        print("mouseHoverEvent")
        super(ButtonItem, self).mouseHoverEvent(ev)

    def hoverEvent(self, ev):       # fix implements - it work
        print("hoverEvent")
        if not self.enabled:
            return
        if ev.enter:
            self.setOpacity(1.0)
        if ev.exit:
            self.setOpacity(0.7)


win = pg.GraphicsLayoutWidget(show=True, title="Button Hover Example")
win.resize(1000, 600)

pg.setConfigOptions(antialias=True)

win.addItem(PlotItem())

if __name__ == '__main__':
    pg.exec()

```

**Comment by hatawong** [comment]
when mouse hover in to the auto button, output:
```
python.exe button_examplt.py
hoverEvent
hoverEvent
hoverEvent
hoverEvent
...
```

**Comment by hatawong** [comment]
call stack
```
hoverEvent, button.py:25
sendHoverEvents, GraphicsScene.py:285
mouseMoveEvent, GraphicsScene.py:196
mouseMoveEvent, GraphicsView.py:364
exec_, __init__.py:407
<module>, button.py:42
```

**Comment by hatawong** [comment]
GraphicsScene.py:285
```
    def sendHoverEvents(self, ev, exitOnly=False):
        ## if exitOnly, then just inform all previously hovered items that the mouse has left.
        
        if exitOnly:
            acceptable=False
            items = []
            event = HoverEvent(None, acceptable)
        else:
            acceptable = not ev.buttons()  ## if we are in mid-drag, do not allow items to accept the hover event.
            event = HoverEvent(ev, acceptable)
            items = self.itemsNearEvent(event, hoverable=True)
            self.sigMouseHover.emit(items)
            
        prevItems = list(self.hoverItems.keys())
            
        for item in items:
            if hasattr(item, 'hoverEvent'):
                event.currentItem = item
                if item not in self.hoverItems:
                    self.hoverItems[item] = None
                    event.enter = True
                else:
                    prevItems.remove(item)
                    event.enter = False
                    
                try:
                    item.hoverEvent(event)
                except:
                    debug.printExc("Error sending hover event:")
```

**Comment by j9ac9k** [comment]
Fixed by #2473
